include/seqan3/core/bit_manipulation.hpp

```diff
@@ -59,4 +68,31 @@ constexpr size_t next_power_of_two(size_t n)
     return n + 1;
 }
 
+/*!\brief Returns the position of the most significant bit (counting from right to left).
+ * \ingroup core
+ *
+ * \param[in] n A unsigned number.
+ *
+ * \attention *n = 0* is a special case and is undefined behaviour.
+ *
+ * \returns The position of the most significant bit.
+ */
+template <std::UnsignedIntegral unsigned_t>
+constexpr uint8_t bit_scan_reverse(unsigned_t n)
```

Comment:
I'm in strong favour of making the position zero-based.
For `00001000` I'd expect `3` as answer, currently it is `4`.

Reply:
What happens if `10000000`, currently we get `0` and we can distinguish that. The current implementation gives you the (zero-based) position of the first 1.

Reply:
changed it